### ISSUE TITLE
Scale to zero

### ISSUE THREAD

**Issue by Y-Less** [hjacobs/kube-aws-autoscaler#39]
Is it possible to use this to scale the nodes down to 0, then back up again?  I did some testing and got the cluster down to 0, but then it just stayed there, even when there were pods queued for scheduling.  Am I doing something else wrong, or will this just not come up from 0?

**Comment by hjacobs** [comment]
@Y-Less where do you run the autoscaler? If it runs as a normal pod on worker nodes it won't run when you scale to zero :smirk:

**Comment by Y-Less** [comment]
I run the autoscaler on the controller node, but scale the workers only.

**Comment by hjacobs** [comment]
@Y-Less you are right, the autoscaler only works if at least one node exists. Reason for this is that it [iterates over all nodes](https://github.com/hjacobs/kube-aws-autoscaler/blob/master/kube_aws_autoscaler/main.py#L214) and only considers these ASGs, i.e. if no node to iterate over => no ASG to set capacity. This could probably be fixed by changing the logic and getting the ASG name from some static configuration or similar (I would want to avoid that).
